ADAL/src/request/ADAcquireTokenSilentHandler.m

```diff
@@ -126,6 +127,12 @@ - (void)acquireTokenByRefreshToken:(NSString*)refreshToken
         request_data[OAUTH2_SCOPE] = _requestParams.scope;
     }
     
+
+    NSString* enrollId = [ADEnrollmentGateway enrollmentIDForHomeUserID:nil userID:(_requestParams.identifier ? _requestParams.identifier.userId : nil)];
```

Comment:
User information in token item is more reliable, as they are sent back from server. 
So i am thinking if we could make it 
`NSString* enrollId = [ADEnrollmentGateway enrollmentIDForHomeUserID:nil userID:(cacheItem.userInformation.userId ? cacheItem.userInformation.userId : _requestParams.identifier.userId)];`